src/Transport/Sending/DefaultOutgoingBatchRouter.cs

```diff
@@ -115,7 +120,32 @@ private string GetViaEntityPathFor(EntityInfo entity)
                 // ASB team promissed to fix the issue with MessagingEntityNotFoundException (missing entity path) - verify that
                 var message = "Failed to dispatch a batch with the following message IDs: " + string.Join(", ", messagesToSend.Select(x => x.MessageId));
                 logger.Error(message, exception);
-                throw;
+
+                // no need to try and send too large messages to another namespace, won't work
+                if (exception is MessageTooLargeException)
+                    throw;
+
+                var fallBackSucceeded = false;
+                if (fallbacks.Any())
+                {
+                    foreach (var fallback in fallbacks)
+                    {
+                        var clones = messagesToSend.Select(x => x.Clone()).ToList();
+                        try
+                        {
+                            await RouteBatchWithEnforcedBatchSizeAsync(fallback, clones).ConfigureAwait(false);
+                            logger.Info("Successfully dispatched a batch with the following message IDs: " + string.Join(", ", clones.Select(x => x.MessageId) + " to fallback namespace"));
+                            fallBackSucceeded = true;
+                            break;
+                        }
+                        catch (Exception ex)
+                        {
+                            logger.Error("Failed to dispatch a batch with the following message IDs: " + string.Join(", ", clones.Select(x => x.MessageId) + " to fallback namespace"), ex);
```

Comment:
We've already logged that error [here](https://github.com/Particular/NServiceBus.AzureServiceBus/pull/129/files#diff-d1b2d15f9b111a00dedb967449c6a732R121). Perhaps this error message should state that we've failed to send sing fallback namespace X?